test/new-e2e/pkg/utils/e2e/client/agent_docker.go

```diff
@@ -0,0 +1,28 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package client
+
+type agentDockerExecutor struct {
+	dockerClient       *Docker
+	agentContainerName string
+}
+
+var _ agentCommandExecutor = &agentDockerExecutor{}
+
+func newAgentDockerExecutor(dockerClient *Docker, agentContainerName string) *agentDockerExecutor {
+	return &agentDockerExecutor{
+		dockerClient:       dockerClient,
+		agentContainerName: agentContainerName,
+	}
+}
+
+func (ae agentDockerExecutor) execute(arguments []string) (string, error) {
+	// We consider that in a container, "agent" is always in path and is the single entrypoint.
+	// It's mostly incorrect but it's what we have ATM.
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution. -->
**❓ question**
Why is this incorrect ?

Reply:
Because you may want to run commands for any agent binary and the different agent binaries are in the same folder.